### PR TITLE
Make last inserted text object work

### PR DESCRIPTION
**Pull request by thalesmello**
Since I'm interested in this functionality, I took the time to make it work.

The original source code threw an error anytime I selected the text object.

So, I've made some changes that make more sense to me.

Such as:

the 'inner' object leaves out the selection leading and trailing whitespace.
the 'around' selects literally what was inserted, including leading and trailing whitespace.

See what you think.

### CODE REVIEW

**Comment by rhysd** [comment]
Thank you for the patch. I'll review.

File: autoload/textobj/lastinserted.vim [hunk 1]
```diff
@@ -1,42 +1,47 @@
...
-    if first[1] > last[1]
-        return 1
-    elseif first[1] == last[1]
-        if first[2] >= last[2]
-            return 1
-        endif
-    endif
-    return 0
+function! s:char_under_cursor()
+    return matchstr(getline('.'), '\%' . col('.') . 'c.')
```
Comment (rhysd) [inline autoload/textobj/lastinserted.vim]: I think you can use `col()` without regex such as `getline('.')[col('.') - 1]`

File: autoload/textobj/lastinserted.vim [hunk 2]
```diff
@@ -1,42 +1,47 @@
...
 endfunction
 
-function! s:select(motion)
-
+function! s:select(mode)
```
Comment (rhysd) [inline autoload/textobj/lastinserted.vim]: Why did you change `motion` param name to `mode`? In Vim, mode generally means Vim's mode such as normal mode, insert mode, visual mode and so on.  So we should avoid using the name for other stuffs.

File: autoload/textobj/lastinserted.vim [hunk 3]
```diff
@@ -1,42 +1,47 @@
...
     if s:invalid_buf()
-        throw "Last inserted buffer is not current buffer."
+        return 0
+    endif
+
+    silent! normal! `.
+
+    if a:mode == 'i' && s:char_under_cursor() == ' '
```
Comment (rhysd) [inline autoload/textobj/lastinserted.vim]: Should we use `\s` regex instead of comparing with `' '`?

File: autoload/textobj/lastinserted.vim (hunk 1 above)
Comment (thalesmello) [inline autoload/textobj/lastinserted.vim]: Please see the new commit.

File: autoload/textobj/lastinserted.vim (hunk 2 above)
Comment (thalesmello) [inline autoload/textobj/lastinserted.vim]: Agreed. Please see the new commit.

File: autoload/textobj/lastinserted.vim (hunk 3 above)
Comment (thalesmello) [inline autoload/textobj/lastinserted.vim]: Modified.

**Comment by rhysd** [comment]
Looks good!  Thank you for nice work 😄
